src/test/java/uk/gov/pay/ledger/expungeorredact/service/ExpungeOrRedactServiceTest.java

```diff
@@ -85,4 +104,88 @@ void shouldObserveJobDurationForMetrics() {
+        when(mockExpungeOrRedactHistoricalDataConfig.getNoOfTransactionsToRedact()).thenReturn(100);
+        when(mockExpungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays()).thenReturn(1);
+
+        when(mockTransactionRedactionInfoDao.getCreatedDateOfLastProcessedTransaction()).thenReturn(parse("2020-01-01T10:15:30Z"));
+
+        TransactionEntity transactionEntity1 = aTransactionFixture().toEntity();
+        TransactionEntity transactionEntity2 = aTransactionFixture().toEntity();
+        when(mockTransactionDao.findTransactionsForRedaction(any(), any(), anyInt())).thenReturn(
+                List.of(transactionEntity1, transactionEntity2)
+        );
+
+        expungeOrRedactService.redactOrDeleteData();
+
+        verify(mockTransactionDao).findTransactionsForRedaction(
+                parse("2020-01-01T10:15:30Z"),
+                parse(SYSTEM_INSTANT).minus(1, DAYS),
+                100);
+
+        verify(mockTransactionDao, times(2)).redactPIIFromTransaction(any());
+        verify(mockEventDao).deleteEventsForTransactions(List.of(transactionEntity1.getExternalId(), transactionEntity2.getExternalId()));
+        verify(mockTransactionRedactionInfoDao).update(transactionEntity2.getCreatedDate());
+
+        verifyNoMoreInteractions(mockTransactionDao);
+        verifyNoMoreInteractions(mockTransactionRedactionInfoDao);
+    }
+
+    @Test
+    void shouldUseCreatedDateOfFirstTransactionWhenInfoIsNotAvailableInTransactionRedactionInfo() {
+        when(mockExpungeOrRedactHistoricalDataConfig.isExpungeAndRedactHistoricalDataEnabled()).thenReturn(true);
+        when(mockExpungeOrRedactHistoricalDataConfig.getNoOfTransactionsToRedact()).thenReturn(100);
+        when(mockExpungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays()).thenReturn(1);
+
+        when(mockTransactionRedactionInfoDao.getCreatedDateOfLastProcessedTransaction()).thenReturn(null);
+        when(mockTransactionDao.getCreatedDateOfFirstTransaction()).thenReturn(Optional.of(parse("2020-01-01T10:15:30Z")));
+
+        expungeOrRedactService.redactOrDeleteData();
+
+        verify(mockTransactionDao).getCreatedDateOfFirstTransaction();
+        verify(mockTransactionRedactionInfoDao).insert(parse("2020-01-01T10:15:30Z"));
+    }
+
+    @Test
+    void shouldUseTheDateDerivedBasedOnConfigForTheStartingIndexWhenNoTransactionsExists() {
+        when(mockExpungeOrRedactHistoricalDataConfig.isExpungeAndRedactHistoricalDataEnabled()).thenReturn(true);
+        when(mockExpungeOrRedactHistoricalDataConfig.getNoOfTransactionsToRedact()).thenReturn(100);
+        when(mockExpungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays()).thenReturn(1);
+
+        when(mockTransactionRedactionInfoDao.getCreatedDateOfLastProcessedTransaction()).thenReturn(null);
+        when(mockTransactionDao.getCreatedDateOfFirstTransaction()).thenReturn(Optional.ofNullable(null));
```

Comment:
Is this a valid scenario? We would never not have a transaction at this point.

Reply:
yeah not in test-prod environments but likely in a local dev environment